File: DesktopBrowser/Server/Utils/Parse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SharpKit.Utils
{
    internal static class Parse
    {
        public static int? TryInt(string s)
        {
            int x;
            if (int.TryParse(s, out x))
                return x;
            return null;
        }
        public static bool? TryBoolean(string s)
        {
            bool x;
            if (bool.TryParse(s, out x))
                return x;
            return null;
        }

        public static long? TryLong(string s)
        {
            long x;
            if (long.TryParse(s, out x))
                return x;
            return null;
        }
    }
}

File: desktopbrowser-server/src/utils/parse.ts
export class Parse {
    static TryInt(s: string): number {
        let x = parseInt(s);
        if (isNaN(x))
            return null;
        return x;
    }
    static TryBoolean(s: string): boolean {
        if (["1", "true", "True", "TRUE"].contains(s))
            return true;
        return false;
    }

    static TryLong(s: string): number {
        return this.TryInt(s);
    }
}

